app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,15 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, except: :index
-  before_action :require_permission, except: :create
+  before_action :require_permission, except: [:create, :repost]
+
+  def repost
+    Post.create(
+      blog: Blog.find(params[:blog_id]),
```

Comment:
I like being pedantic here and doing `:blog`, `:body`, `:user`, but to save a database I'm guessing I can swap between the objects and their ids? ie, do `blog_id: params[:blog_id]` then `body` and `user`.

Reply:
I'm not sure if you saw my comment on the whole PR that talked about giving credit to the original poster, but this would sort of address your concerns.

In which case 1. Could be solved by something like
`Posts.all.where(original_post_id: @post.id)` and there's a query to collect all users from that list but can't construct it now at lunch :P
And 2. Could be solved by
`Post.find(@post.original_post_id).user`